yarn-project/acir-simulator/src/client/client_execution_context.ts

```diff
@@ -127,28 +112,35 @@ export class ClientTxExecutionContext {
     offset: number,
     returnSize: number,
   ): Promise<ACVMField[]> {
-    const pendingNotes = this.getPendingNotes(contractAddress, storageSlot, sortBy, sortOrder, limit);
+    const storageSlotField = fromACVMField(storageSlot);
 
-    const dbNotes = await this.db.getNotes(
-      contractAddress,
-      fromACVMField(storageSlot),
-      sortBy.map(field => +field),
-      sortOrder.map(field => +field),
-      limit,
+    // TODO(https://github.com/AztecProtocol/aztec-packages/issues/920): don't 'get' notes nullified in pendingNullifiers
+    const pendingNotes = this.pendingNotes.filter(
+      n => n.contractAddress.equals(contractAddress) && n.storageSlot.equals(storageSlotField),
     );
 
-    // TODO: Sort again.
-    const notes = [...pendingNotes, ...dbNotes].slice(offset, limit ? offset + limit : undefined);
+    const dbNotes = await this.db.getNotes(contractAddress, storageSlotField);
```

Comment:
So will this just get ALL notes from the DB that match contractAddress & slot?

Reply:
Yeah, we have to do that anyway :( Because the preimage is stored as a single field in the db. So we always need to get all notes matching the contractAddress and slot, to parse it, sort it, and slice it.